mantaray/received.py

```diff
@@ -312,13 +327,27 @@ def _handle_cap(server_view: views.ServerView, args: list[str]) -> None:
     if subcommand == "ACK":
         acknowledged = set(args[-1].split())
         if "sasl" in acknowledged:
+            server_view.core.cap_list.add("sasl")
             server_view.core.send("AUTHENTICATE PLAIN")
+        if "away-notify" in acknowledged:
+            server_view.core.cap_list.add("away-notify")
     elif subcommand == "NAK":
         rejected = set(args[-1].split())
         if "sasl" in rejected:
             # TODO: this good?
             raise ValueError("The server does not support SASL.")
 
+    server_view.core.number_of_cap_req -= (
+        1  # To evaluate how many more ACK/NAKs will be received from server
+    )
+
+    # Need to wait for login success/fail from server if sasl is enabled before sending "CAP END"
```

Comment:
I don't understand this comment. What exactly is "enabling" SASL? Why do we need to wait? What happens if we don't wait?